compat/mingw.c

```diff
@@ -383,6 +383,21 @@ static wchar_t *normalize_ntpath(wchar_t *wbuf)
 	return wbuf;
 }
 
+static int do_unlink(const char *pathname, const wchar_t *wpathname,
+		     int (*unlinker)(const wchar_t *))
+{
+	char temppath[MAX_PATH];
+	wchar_t wtemppath[MAX_LONG_PATH];
+	strncpy(temppath, pathname, sizeof(temppath) - 8);
+	strcat(temppath, ".XXXXXX");
+	mktemp(temppath);
+	if (xutftowcs_long_path(wtemppath, temppath) >= 0) {
+		if (MoveFileExW(wpathname, wtemppath, MOVEFILE_REPLACE_EXISTING | MOVEFILE_COPY_ALLOWED))
+			return unlinker(wtemppath);
+	}
+	return unlinker(wpathname);
```

Comment:
If either `xutftowcs_long_path` or `MoveFileExW` fail, then we are in an unknown bad state, and trying to continue will obfuscate the issue. Instead we should return an error and give the caller a chance to report it. If it turns out later that there is a special case which we can handle better, then we can do so while explicitly checking the corresponding error code.